torch/_inductor/runtime/triton_heuristics.py

```diff
@@ -1003,7 +1003,9 @@ def save_gpu_kernel(self, stream, launcher):
         bin_type = {"hip": "hsaco", "xpu": "spv"}.get(self.device_props.type, "cubin")
         binary = launcher.bin.asm[bin_type]
         # Also store asm code which can be used for debugging and generating cpp package
-        asm_type = {"hip": "amdgcn", "cuda": "ptx"}.get(self.device_props.type, None)
+        asm_type = {"hip": "amdgcn", "cuda": "ptx", "xpu": "spv"}.get(
```

Comment:
Since we already generate spv for xpu, shouldn't the added new option be a no-op for xpu?

Reply:
Hi @desertfire , Yes, in theory XPU doesn’t need this option. I added it to align with the user interface, and also to fix some errors that occur on XPU when this option is specified in UT. For example:  https://hud.pytorch.org/pr/pytorch/pytorch/147693#43085076369
```
RuntimeError: Failed to run autotuning code block: Missing kernel assembly code
```

And this may also helps avoid adding multiple if-else checks in the code specifically for XPU.